api/libs/scenarios-planning-unit/src/domain/lock-planning-units.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { lockPlanningUnits } from './lock-planning-units';
+
+const fixtures = {
+  availablePlanningUnits: ['1', '2', '3', '4'],
+  someOfTheAvailable: ['1', '4'],
+  withNonAvailable: ['1', '5'],
+};
+
+test.each([
+  [fixtures.availablePlanningUnits, fixtures.availablePlanningUnits, 0],
+  [fixtures.someOfTheAvailable, fixtures.availablePlanningUnits, 0],
+  [fixtures.withNonAvailable, fixtures.availablePlanningUnits, 1],
+  [[], fixtures.availablePlanningUnits, 0],
+  [fixtures.someOfTheAvailable, [], fixtures.someOfTheAvailable.length],
+])(
+  `when changing %p out of %p, it returns %p errors`,
+  (unitsToChange, availableUnits, errorsCount) => {
+    expect(
+      lockPlanningUnits(unitsToChange, availableUnits).errors.length,
+    ).toEqual(errorsCount);
+  },
+);
```

Comment:
while the test makes sense, the ambiguity of the `lockPlanningUnits()` function (see comment below), or my misunderstanding if it, seems to make this test not so obvious - aka I am not sure what exactly we want to test here.

- `lockPlanningUnits()` doesn't seem to do what the name suggests (see below)
- I'd use different ids (even just an array of four UUIDv4s, a subset of two elements of this array, and a strict subset + another generated UUID) to avoid any possible confusion with the lock status values, which use similar numbers to the ones in the fixtures

Reply:
@hotzevzl fixed in https://github.com/Vizzuality/marxan-cloud/pull/224/commits/8b9c4a7bbed5d99bc818bedd331199dc29415f8e - more verbose names, added named types as well